Softeq.XToolkit.PushNotifications.Droid/XFirebaseMessagingService.cs

```diff
@@ -4,28 +4,35 @@
 using System;
 using Android.App;
 using Firebase.Messaging;
+using Softeq.XToolkit.PushNotifications.Abstract;
+using Softeq.XToolkit.PushNotifications.Droid.Services;
+using Softeq.XToolkit.WhiteLabel;
 
 namespace Softeq.XToolkit.PushNotifications.Droid
 {
     [Service(Exported = true)]
     [IntentFilter(new[] { "com.google.firebase.MESSAGING_EVENT" })]
-    public class XFirebaseMessagingService : FirebaseMessagingService
+    internal class XFirebaseMessagingService : FirebaseMessagingService
     {
-        public static Action<string>? OnTokenRefreshed;
-        public static Action<RemoteMessage>? OnNotificationReceived;
+        private readonly IDroidFirebaseMessagingHandler _messagingHandler;
+
+        public XFirebaseMessagingService()
+        {
+            _messagingHandler = Dependencies.Container.Resolve<DroidPushNotificationsService>();
```

Comment:
looks like should be `IDroidFirebaseMessagingHandler`

Reply:
Here are my thoughts
`IDroidFirebaseMessagingHandler` doesn't make sense to make it public, so it can't be registered in a container.
Right now the `DroidPushNotificationsService` is (or should be) registered in the container like this
```
builder.Singleton<DroidPushNotificationsService>();
builder.Singleton<IPushNotificationsService>(container => container.Resolve<DroidPushNotificationsService>());
builder.Singleton<IActivityLauncherDelegate>(container => container.Resolve<DroidPushNotificationsService>());
```
So we can get the `DroidPushNotificationsService` from the container and use it internally as a message handler from the `FirebaseMessagingService`, without showing it to the outside